.github/workflows/nightly_smoke_tests.yml

```diff
@@ -0,0 +1,98 @@
+name: Nightly Smoke Tests
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+    inputs:
+      sha:
+        description: 'Commit SHA to test'
+        required: false
+        default: ''
+        type: string
+
+jobs:
+  smoke_tests:
+    if: github.repository == 'linode/linodego' || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: 'recursive'
+          ref: ${{ github.event.inputs.sha || github.ref }}
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.x'
+
+      - name: Run smoke tests
+        id: smoke_tests
+        run: |
+          make smoketest
+        env:
+          LINODE_TOKEN: ${{ secrets.DX_LINODE_TOKEN }}
+
+      - name: Notify Slack
+        if: always()
```

Comment:
Would it make sense to ensure that the repo is `linode/linodego` here to cover the case where `workflow_dispatch` is used on a fork?